crates/service/src/status.rs

```diff
@@ -0,0 +1,108 @@
+//! Status handle.
+
+use std::{any::Any, fmt::Debug};
+
+use serde::de::DeserializeOwned;
+use tokio::sync::watch;
+
+use crate::Service;
+
+/// Generic boxed status value which can be downcast to a concrete type.
+pub type AnyStatus = Box<dyn Any + Sync + Send + 'static>;
+
+/// Service status monitor handle.
+#[derive(Clone, Debug)]
+pub struct ServiceMonitor<S: Service> {
+    status_rx: watch::Receiver<S::Status>,
+}
+
+impl<S: Service> ServiceMonitor<S> {
```

Comment:
Please pull the whole crate, it's missing the most recent changes (from **6aab219** commit)